src/messages/MOSDOp.h

```diff
@@ -242,14 +236,14 @@ struct ceph_osd_request_head {
       ::encode_nohead(snaps, payload);
     } else {
       header.version = HEAD_VERSION;
-      ::encode(client_inc, payload);
+      ::encode(pgid, payload);
```

Comment:
One thing about reordering: Why exactly we can't do like this? New version (6) is coded in MOSDOp.h, and clients using RADOS will encode messages in new version. For kernel RBD module (which encoded version 4) backward compatibility for splitted decoding is implemented.